### PR TITLE
Build mock if there's no base templates

### LINKED ISSUE #47: Build mock if not templates
If no template need build bundle without `bem-xjst` core. Instead of it need provide mocks.

This can be useful to build BEMHTML for run in browsers with depsByTech.

For BEMHTML `apply` method should always returns empty string:

```js
BEMHTML.apply = function () { return ''; };
```

For BEMTREE `apply` method should always returns empty object:

```js
BEMTREE.apply = function () { return Vow.resolve({}); };
```

### PR DESCRIPTION
**Pull request by blond**
Resolved #47

### CODE REVIEW

File: test/techs/bemhtml/bemhtml.test.js [hunk 1]
```diff
@@ -28,6 +28,17 @@ describe('bemhtml', function () {
         mock.restore();
     });
 
+    it('must generate mock if not templates', function () {
```
Comment (tadatuta) [inline test/techs/bemhtml/bemhtml.test.js]: if not -> if there is no

File: test/techs/bemtree.test.js [hunk 2]
```diff
@@ -64,6 +64,16 @@ describe('bemtree', function () {
             });
     });
 
+    it('must generate mock if not templates', function () {
```
Comment (tadatuta) [inline test/techs/bemtree.test.js]: same

File: techs/bemhtml.js [hunk 3]
```diff
@@ -37,6 +40,13 @@ module.exports = require('./bem-xjst').buildFlow()
     .defineOption('modulesDeps')
     .useFileList(['bemhtml', 'bemhtml.xjst'])
     .builder(function (sourceFiles) {
+        if (sourceFiles.length === 0) {
```
Comment (tavriaforever) [inline techs/bemhtml.js]: maybe `if (!sourceFiles.length)` ?

File: techs/bemtree.js [hunk 4]
```diff
@@ -35,6 +38,14 @@ module.exports = require('./bem-xjst').buildFlow()
     .defineOption('modulesDeps')
     .useFileList(['bemtree'])
     .builder(function (sourceFiles) {
+        if (sourceFiles.length === 0) {
```
Comment (tavriaforever) [inline techs/bemtree.js]: and here maybe if (!sourceFiles.length) ?

**Comment by tavriaforever** [comment]
:muscle:

File: techs/bemhtml.js (hunk 3 above)
Comment (blond) [inline techs/bemhtml.js]: No explicit is better than implicit ;)

**Pull request merged**